airbyte-integrations/bases/base-normalization/normalization/transform_catalog/utils.py

```diff
@@ -16,76 +17,125 @@ def remove_jinja(command: str) -> str:
     return str(command).replace("{{ ", "").replace(" }}", "")
 
 
+# transform oneOf nested list ot plain types list
+def get_plain_list_from_one_of_array(one_of_definition) -> list:
+    one_of_property = one_of_definition[data_type.ONE_OF_VAR_NAME]
+    plain_datatypes_list = list()
+    for prop in one_of_property:
+        plain_datatypes_list.append(prop[data_type.REF_TYPE_VAR_NAME])
+    return plain_datatypes_list
+
+
 def is_string(property_type) -> bool:
-    return property_type == "string" or "string" in property_type
+    if data_type.ONE_OF_VAR_NAME in property_type and data_type.ONE_OF_VAR_NAME in property_type:
+        plain_datatypes_list = get_plain_list_from_one_of_array(property_type)
+        return plain_datatypes_list == data_type.STRING_TYPE or data_type.STRING_TYPE in plain_datatypes_list
+    else:
+        return property_type == data_type.STRING_TYPE or data_type.STRING_TYPE in property_type
+
+
+def is_binary_datatype(property_type) -> bool:
+    if data_type.ONE_OF_VAR_NAME in property_type and data_type.ONE_OF_VAR_NAME in property_type:
+        plain_datatypes_list = get_plain_list_from_one_of_array(property_type)
+        return plain_datatypes_list == data_type.BINARY_DATA_TYPE or data_type.BINARY_DATA_TYPE in plain_datatypes_list
+    else:
+        return property_type == data_type.BINARY_DATA_TYPE or data_type.BINARY_DATA_TYPE in property_type
```

Comment:
This was just a generic verification to check for object. As depends on the place where this is called, It may be a single object, it may be a dict, or it may be inside of oneOf. But if we have an assumption that binary datatype could not be a part of oneOf (as per your comment above) - then yes,  this may be simplified.

Reply:
@edgao This is what I mean by saying that some of the methods may be called from different places and sometimes passing a simple object and sometimes a "dict". This was also mindblowing for me after strict types in java :) 


Here is an example where it's passed like just a string. That is why some checks may look redundant at the first glance

![Selection_139](https://user-images.githubusercontent.com/4688354/206731139-aad18b76-62e0-47b2-9cef-5bb4cd0a9d27.png)

Reply:
hmmmmmmmmmmmmmmmm

how hard would it be to make all of these methods _always_ expect a dict? It feels like this is causing us a lot of confusion :/ and this seems like a good opportunity to clean up the code a little bit